MR4/expression/call.3.mr

```diff
@@ -0,0 +1,99 @@
+## MR4 compiler - Call expression
+
+## Function call expression
+class Call-expression(Expression)
+  owner Expression function
+  owner Call-arguments arguments
+  
+  func inst parse-new(user String ends):(
+      owner Expression expression, copy Char end)
+    new Call-expression call-expression
+    call-expression.parse(owner expression):(copy end)
+    expression := call-expression
+  
+  func inst parse(owner Expression function):(copy Char end)
+    self.function := function
+    new Call-arguments in self.arguments
+    self.arguments.parse():(copy end)
+  
+  func dynamic write()
+    self.function.write()
+    self.arguments.write()
+
+
+static Call-arguments
+  owner Call-argument parameters
+  owner Call-argument outputs
+  
+  ## parsing `(access value, ...):(access value, ...)`
+  func inst parse():(copy Char end)
+    self.parse-args():(owner self.parameters, copy end)
+    if end = ':'
+      read-expect(user "(")
+      self.parse-args():(owner self.outputs, copy end)
+  
+  ## parsing single argument list `(access value, ...)`
+  func inst parse-args():(owner Call-argument first, copy Char end)
+    user Call-argument arg
+    var String{128} access-str
```

Comment:
This can be a lot shorter, because the maximum size is just the maximum access keyword, right?